File: summer/WebContent/js/system/windows/IContentHost.js
/**
 * IContentHost
 */

define(["dojo/_base/declare", "system/Type"], function(declare, Type){
	var IContentHost = declare("IContentHost", null,{
//		IInputElement 
		InputHitTest:function(/*Point*/ point){},
//		ReadOnlyCollection<Rect> 
		GetRectangles:function(/*ContentElement*/ child){},
//		void 
		OnChildDesiredSizeChanged:function(/*UIElement*/ child){}
	});
	
	Object.defineProperties(IContentHost.prototype, {
//		IEnumerator<IInputElement> 
		HostedElements:
		{
			get:function(){}
		}
	});
	
	IContentHost.Type = new Type("IContentHost", IContentHost, [Object.Type]);
	return IContentHost;
});

File: org.summer.view.widget/src/org/summer/view/window/IContentHost.java
package org.summer.view.window;

import org.summer.view.widget.ContentElement;
import org.summer.view.widget.IInputElement;
import org.summer.view.widget.Point;
import org.summer.view.widget.Rect;
import org.summer.view.widget.UIElement;
import org.summer.view.widget.collection.IEnumerator;
import org.summer.view.widget.collection.ReadOnlyCollection;

public interface IContentHost {
	ReadOnlyCollection<Rect> GetRectangles(ContentElement child);

	IInputElement InputHitTest(Point point);

	void OnChildDesiredSizeChanged(UIElement child);
	
	IEnumerator<IInputElement> HostedElements { get; };
}
